diff --git a/python/google/protobuf/internal/decoder.py b/python/google/protobuf/internal/decoder.py
index 31869e4..327a756 100755
--- a/python/google/protobuf/internal/decoder.py
+++ b/python/google/protobuf/internal/decoder.py
@@ -165,6 +165,10 @@ _DecodeSignedVarint = _SignedVarintDecoder((1 << 64) - 1, long)
 _DecodeVarint32 = _VarintDecoder((1 << 32) - 1, int)
 _DecodeSignedVarint32 = _SignedVarintDecoder((1 << 32) - 1, int)
 
+def _EnsureInt(value):
+  if isinstance(value, int):
+    return value
+  return ord(value)
 
 def ReadTag(buffer, pos):
   """Read a tag from the buffer, and return a (tag_bytes, new_pos) tuple.
@@ -178,7 +182,7 @@ def ReadTag(buffer, pos):
   """
 
   start = pos
-  while six.indexbytes(buffer, pos) & 0x80:
+  while _EnsureInt(six.indexbytes(buffer, pos)) & 0x80:
     pos += 1
   pos += 1
   return (buffer[start:pos], pos)
